### PR TITLE
GEN-1645 - feat(ProductReviews): added trustpilot reviews

### PR DESCRIPTION
**Pull request by guilhermespopolin**
## Describe your changes

* Updates `fetchTrustpilotData` function so it receives a `locale`. We're gonna use that to fetch reviews for the correct language.
* Adds a method for fetching _Trustpilot_ reviews on `trustpilot` service
* `ReviewComment`
  * Removed `maxHeight` constraint as _Trustpilot_ reviews can be bigger them _Product_ reviews and that constraint might results on text overflowing review's card.
  * Show _author_ accordingly based on `type` prop: For `company` reviews we shown _Trustpilot_ logo. For `product`, we show _Verified customer_ label.
* `ProductReviews`: updated so it supports _Trustpilot_ reviews. It ended up with some copied & pasted sections on the UI that will be handled down in the stack.
* Refactoring:
 * Change `Score` type so it uses `number` instead of `string`
 * Update `Stars` and `ReviewsDistributionByScore` API so they receive a `score` prop in a `number` format. 

image

Can be tested [here](https://hedvig-dot-com-git-gen-1645-feattruspilot-comments-hedvig.vercel.app/se/forsakringar/olycksfallsforsakring/av)

## Justify why they are needed

Desired by Peter for getting first release of _ProductReviews_ out.

### CODE REVIEW

**Comment by guilhermespopolin** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3791?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `main`
  * **PR #3789** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3789?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #3790** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3790?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #3791** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3791?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3791?utm_source=stack-comment).

File: apps/store/src/components/ProductReviews/ProductReviews.tsx [hunk 1]
```diff
@@ -86,51 +90,65 @@ export const ProductReviews = (props: Props) => {
...
+                <Space y={1}>
+                  {averageRating && trustpilotData && (
+                    <SpaceFlex space={0.5}>
+                      <Button
+                        value={Tabs.PRODUCT}
+                        size="medium"
+                        variant={selectedTab === Tabs.PRODUCT ? 'primary-alt' : 'secondary'}
+                        fullWidth={true}
+                        onClick={handleTabChange}
+                      >{`Omdömen (${averageRating.score})`}</Button>
```
Comment (robinandeer) [inline apps/store/src/components/ProductReviews/ProductReviews.tsx:123]: are we planning to lokalise this? perhaps add a comment

**Review by robinandeer — APPROVED** [review]
Very nice refactor -- much cleaner and fewer type conversions

**Review by gustaveen — APPROVED** [review]
Great job!

File: apps/store/src/components/ProductReviews/ProductReviews.tsx (hunk 1 above)
Comment (guilhermespopolin) [reply apps/store/src/components/ProductReviews/ProductReviews.tsx:123]: Yes, will do.

**Comment by guilhermespopolin** [comment]
### Merge activity

* **Jan 2, 9:53 AM**: @guilhermespopolin started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3791).

**Pull request merged**